backend/apps/matching/tests.py

```diff
@@ -1,14 +1,56 @@
 from io import StringIO
 
 from django.core.management import call_command
-from django.test import TestCase
+from django.test import TestCase, TransactionTestCase
+
+
+def create_superuser():
+    output = StringIO()
+    args = ["--noinput", "--email=staff@example.com", "--username=staff"]
+    opts = {}
+    call_command("createsuperuser", *args, stdout=output, **opts)
+    return output.getvalue()
+
+
+def create_fakeusers(a=10, b=5):
+    output = StringIO()
+    args = ["--add-a=" + str(a), "--add-b=" + str(b)]
+    opts = {}
+    call_command("createfakeusers", *args, stdout=output, **opts)
+    output_string = output.getvalue()
+    assert ("Created " + str(b) + " participants") in output_string
+    return output_string
 
 
 class ProvenBugsTestCase(TestCase):
     def test_superuser_creation(self):
 
+        output = create_superuser()
+        self.assertIn("Superuser created successfully", output)
+
+
+class MigrationTestCase(TransactionTestCase):
+    def test_migration_with_data(self):
+        # Migrate to starting point
         output = StringIO()
-        args = ["--noinput", "--email=staff@example.com", "--username=staff"]
-        opts = {}
-        call_command("createsuperuser", *args, stdout=output, **opts)
-        self.assertIn("Superuser created successfully", output.getvalue())
+        args = ["matching", "0001_initial"]
```

Comment:
Could we maybe create a code test if the permission migration is still present? That way we would prevent accidental deletions. But no must have.

Reply:
Nice idea. Just a `file-exists`-test? But actually the only improvement I can see would be the more specific error message, since the test you linked test would automatically fail too :)